### PR TITLE
TF-3087 Fix no suggestion for email address in search bar on mobile

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3087

## Resolved

- Mobile:

[Screen_recording_20240830_144320.webm](https://github.com/user-attachments/assets/d6ac4740-c45a-44e7-8ab2-e0ba6cf9d081)

- Web:

https://github.com/user-attachments/assets/b75487cd-afdf-4c0d-935a-7da44de18d59

### CODE REVIEW

**Comment by tddang-linagora** [comment]
@tddang-linagora Test on mobile:

- [ ] Android
- [ ] iOS

**Review by tddang-linagora — APPROVED** [review]

File: core/lib/presentation/views/quick_search/quick_search_input_form.dart [hunk 1]
```diff
@@ -1200,65 +1200,10 @@ class _SuggestionsListState<T, P, R> extends State<_SuggestionsList<T, P, R>>
...
-            onTap: () => widget.onContactSuggestionSelected?.call(contact),
-          ),
-        );
-      } else {
-        return const SizedBox.shrink();
-      }
-    }).toList() ?? [];
+    final listItemSuggestionWidget = _buildListViewSuggestionWidget();
+    final loadingWidget = _buildLoadingBarWidget();
+    final listAction = _buildListActionWidget();
```
Comment (hoangdat) [inline core/lib/presentation/views/quick_search/quick_search_input_form.dart:1205]:
```suggestion
    final listActionWidget = _buildListActionWidget();
```

File: core/lib/presentation/views/quick_search/quick_search_input_form.dart [hunk 2]
```diff
@@ -1297,55 +1239,9 @@ class _SuggestionsListState<T, P, R> extends State<_SuggestionsList<T, P, R>>
...
-                child: widget.actionButtonBuilder!(context, action),
-              ),
-            );
-          } else {
-            return const SizedBox.shrink();
-          }
-        }).toList());
+    final listItemRecent = _buildListViewRecentWidget();
+    final loadingWidget = _buildLoadingBarWidget();
+    final listAction = _buildListActionWidget();
```
Comment (hoangdat) [inline core/lib/presentation/views/quick_search/quick_search_input_form.dart:1244]:
```suggestion
    final listActionWidget = _buildListActionWidget();
```

**Comment by hoangdat** [comment]
https://github.com/user-attachments/assets/0f0f0c4c-2e96-4620-b2dc-a47cc9855ed2

- [ ] we should add loading for screen (from 0:04s ~)
- [ ] we should show no result view in suggestion also

File: core/lib/presentation/views/quick_search/quick_search_input_form.dart (hunk 2 above)
Comment (dab246) [reply core/lib/presentation/views/quick_search/quick_search_input_form.dart]: done

File: core/lib/presentation/views/quick_search/quick_search_input_form.dart (hunk 1 above)
Comment (dab246) [reply core/lib/presentation/views/quick_search/quick_search_input_form.dart]: done

**Comment by dab246** [comment]
> Screenrecorder-2024-09-13-10-01-47-482.mp4 
> * [x]  we should add loading for suggestion screen (from 0:04s ~). Only for mobile, because it is separate screen, not the same with web

Fixed. Please see below demo:


[Screen_recording_20240913_171255.webm](https://github.com/user-attachments/assets/0b772db6-7fcc-45d8-9984-e7055c32e6a2)

**Review by hoangdat — APPROVED** [review]

**Review by tddang-linagora — APPROVED** [review]

**Pull request merged**